feat: Add Members Detail & New

suinj8 opened the pull request:
구성원 - 교수, 연구원, 석사 , 학사, 운영위원회
각각 상세보기, 등록하기 구현하였습니다.

상세보기(교수)
![image](https://user-images.githubusercontent.com/90179555/185892729-4d76f4d7-14be-4874-ba33-fd6747fc3904.png)

클릭 시
![image](https://user-images.githubusercontent.com/90179555/185892811-76c3ee53-bd8b-481c-9e21-02e68114f76c.png)
위처럼 보여지게 됩니다.
useParams를 이용하여 클릭된 row의 id값을 전달받아 그 값으로 api요청을 보내고 응답을 받아
데이터를 선 출력하는 방식입니다.

우선은 MockAPI로 구현하였기 때문에 시간, 공간상 모두 만들 수 없어서
교수페이지에만 Mock API를 이용하여 출력되게 해놓았고
나머지 페이지는 주석처리 해두었습니다. (틀만 보여지고 데이터는 나오지 않습니다)

등록하기
저번 PR과 동일한 형태로 구성하였습니다.

전부 비슷한 형식이지만 API구성에 따라 지울건 지우고 추가할건 추가하였습니다.
내용이 많아서 꼼꼼히 한다고 했지만 실수가 있을 수 있으니 리뷰하시면서 오류 확인 부탁드립니다.

suinj8 committed:
feat: Add Members Detail & New

구성원 - 교수, 연구원, 석사 , 학사, 운영위원회
각각 자세히보기, 등록하기 구현

LimEunSang commented on src/admin/components/member/member_page/ProfessorDetail.js:
```diff
@@ -0,0 +1,202 @@
+  const [phoneNum, setPhoneNum] = useState("");
+  const [ID, setID] = useState("");
+  const [password, setPassword] = useState("");
+  const [img, setImg] = useState("");
```
저는 멤버의 정보를 저장하는 객체를 이렇게 하나하나 선언 안하고 하나의 객체로 관리하는 방법을 썼었는데
뭐가 정답이고 더 좋을지는 잘 모르겠지만 참고용으로 제 코드를 첨부할게요

주석처리로 게시물 데이터 관리 객체 부분을 보시면 하나의 객체로 선언했습니다.
![image](https://user-images.githubusercontent.com/86942472/186156279-ec40d061-15f8-40ae-9e88-e75ef05aaced.png)

suinj8 replied:
새롭게 제시해주신 방법이 더 깔끔할것 같네요.
단점이라면 한번접근에 2번의 경로를 거쳐야해서 느려진다 인데
사실상 느끼지 못할 수준이니까 더 깔끔해보이는 아래가 좋아보입니다.
다른분들 의견도 은상님 방법이 깔끔하다고 생각하신다면 
바꿔서 커밋해놓겠습니다.  (아마 전에 머지한 은영님 코드도 다 바꿔야 할거에요)

euny0ung replied:
저런 방법이 있었군요 저도 저게 더 깔끔해보이니 수정하겠습니다.

suinj8 replied:
수정완료하였습니다.
state값 변경 코드가 길어지긴 했지만 코드자체는 더 직관적으로 바뀐거 같네요.

baaaam771 replied:
저도 아래 방식이 읽기 편해보여요

LimEunSang reviewed (approved):
고생하셨습니다.

approve 이지만 리뷰 댓글 한 번 확인해주세요~

나머진 오류 없어보입니다~

LGTM!

suinj8 committed:
feat: refactoring state

객체 선언, 데이터 저장방식 변경

suinj8 committed:
feature : Update Admin Professor

Grid를 감싸는 div에 고정 높이를 제공하여
Data grid에 스크롤이 생기도록 하였습니다.
Data grid 특성상 무한으로 받아오는 기능은 없다고 하네요.
따라서 100개씩 보여주는 방식을 택하고 기본적으로 제공해주는
UI를 사용하는것이 좋아보입니다.

suinj8 committed:
feat: Update Admin Professor

data virtualization을 이용해서 한페이지에 전부 출력하려고 했지만
data grid의 한계로 불가능 합니다.
따라서 한 페이지에 최대 100개로 출력하는것이 나아보입니다.
datagrid를 감싸는 div태그의 높이를 고정크기로 하여  데이터를 집어넣으면
자동으로 datagrid의 스크롤이 생기는 것(최대 100개)을 이용하였습니다.

추가적으로 많은 데이터를 넣어보려고 지저분하게 dummy를 사용했습니다.
양해 부탁드립니다. 나중에 의견 취합되고 확인되면 다시 지우고 커밋 후에
머지하도록 하겠습니다.
